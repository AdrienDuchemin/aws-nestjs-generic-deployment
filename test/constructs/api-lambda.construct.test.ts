import { expect as expectCDK, SynthUtils, countResources, haveResource } from '@aws-cdk/assert'
import { Code } from '@aws-cdk/aws-lambda'
import { Stack } from '@aws-cdk/core'
import { ApiLambda, LAMBDA_API_TIMEOUT } from '../../src/constructs/api-lambda.construct'

describe('ApiLambda', () => {
  let stack: Stack

  beforeAll(()=> {
    stack = new Stack()
    new ApiLambda(stack, 'ApiLambda', {
      lambdaProps: {
        code: Code.fromInline('lambda'),
        handler: 'handler',
      }
    })
  })

  it('should create one Lambda', () => {
    expectCDK(stack).to(countResources("AWS::Lambda::Function", 1))
    expectCDK(stack).to(haveResource("AWS::Lambda::Function",{
      Timeout: LAMBDA_API_TIMEOUT,
    }))
  })

  it('should create one Api Gateway', () => {
    expectCDK(stack).to(countResources("AWS::ApiGateway::RestApi", 1))
  })

  it('should not have changed', () => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
  })
})