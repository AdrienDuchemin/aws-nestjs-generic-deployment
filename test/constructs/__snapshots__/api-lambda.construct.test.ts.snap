// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiLambda should not have changed 1`] = `
Object {
  "Outputs": Object {
    "ApiLambdaLambdaApiLambdaRestApiEndpoint88F2A377": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "ApiLambdaLambdaApiLambdaRestApiDeploymentStageprod4AC10FC5",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Resources": Object {
    "ApiLambdaLambdaApiLambdaRestApiANY840CBB57": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "ApiLambdaLambdaBA627534",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "ApiLambdaLambdaApiLambdaRestApiANYApiPermissionApiLambdaLambdaApiLambdaRestApi65320199ANYB8177B2B": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaLambdaBA627534",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
              },
              "/",
              Object {
                "Ref": "ApiLambdaLambdaApiLambdaRestApiDeploymentStageprod4AC10FC5",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaLambdaApiLambdaRestApiANYApiPermissionTestApiLambdaLambdaApiLambdaRestApi65320199ANY96B68574": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaLambdaBA627534",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaLambdaApiLambdaRestApiAccount07BF97E7": Object {
      "DependsOn": Array [
        "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaLambdaApiLambdaRestApiCloudWatchRole9AD5139D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "ApiLambdaLambdaApiLambdaRestApiCloudWatchRole9AD5139D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ApiLambdaLambdaApiLambdaRestApiDeploymentD9F6CB4Eddb87a546bf297b947f42c33ea75033a": Object {
      "DependsOn": Array [
        "ApiLambdaLambdaApiLambdaRestApiproxyANY14E5BCAF",
        "ApiLambdaLambdaApiLambdaRestApiproxyE016477E",
        "ApiLambdaLambdaApiLambdaRestApiANY840CBB57",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "ApiLambdaLambdaApiLambdaRestApiDeploymentStageprod4AC10FC5": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "ApiLambdaLambdaApiLambdaRestApiDeploymentD9F6CB4Eddb87a546bf297b947f42c33ea75033a",
        },
        "RestApiId": Object {
          "Ref": "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "ApiLambdaLambdaApiLambdaRestApiE4A5155A": Object {
      "Properties": Object {
        "Name": "ApiLambdaRestApi",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "ApiLambdaLambdaApiLambdaRestApiproxyANY14E5BCAF": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "ApiLambdaLambdaBA627534",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "ApiLambdaLambdaApiLambdaRestApiproxyE016477E",
        },
        "RestApiId": Object {
          "Ref": "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "ApiLambdaLambdaApiLambdaRestApiproxyANYApiPermissionApiLambdaLambdaApiLambdaRestApi65320199ANYproxyCDCA00C0": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaLambdaBA627534",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
              },
              "/",
              Object {
                "Ref": "ApiLambdaLambdaApiLambdaRestApiDeploymentStageprod4AC10FC5",
              },
              "/*/{proxy+}",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaLambdaApiLambdaRestApiproxyANYApiPermissionTestApiLambdaLambdaApiLambdaRestApi65320199ANYproxy3E492C3D": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaLambdaBA627534",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
              },
              "/test-invoke-stage/*/{proxy+}",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaLambdaApiLambdaRestApiproxyE016477E": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": Object {
          "Ref": "ApiLambdaLambdaApiLambdaRestApiE4A5155A",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "ApiLambdaLambdaBA627534": Object {
      "DependsOn": Array [
        "ApiLambdaLambdaServiceRole56C749CD",
      ],
      "Properties": Object {
        "Code": Object {
          "ZipFile": "lambda",
        },
        "FunctionName": "ApiLambdaLambda",
        "Handler": "handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaLambdaServiceRole56C749CD",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "ApiLambdaLambdaServiceRole56C749CD": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
